satellite/marketing/marketingweb/server.go

```diff
@@ -102,46 +146,119 @@ func (s *Server) parseTemplates() (err error) {
 		filepath.Join(s.templateDir, "free-offers-modal.html"),
 	)
 
+	pageNotFoundFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "page-not-found.html"),
+	)
+
+	internalErrorFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "internal-server-error.html"),
+	)
+
+	badRequestFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "err.html"),
+	)
+
 	s.templates.home, err = template.New("landingPage").ParseFiles(homeFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	pageNotFoundFiles := append(s.commonPages(),
-		filepath.Join(s.templateDir, "page-not-found.html"),
-	)
-
 	s.templates.pageNotFound, err = template.New("page-not-found").ParseFiles(pageNotFoundFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	internalErrorFiles := append(s.commonPages(),
-		filepath.Join(s.templateDir, "internal-server-error.html"),
-	)
-
 	s.templates.internalError, err = template.New("internal-server-error").ParseFiles(internalErrorFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
+	s.templates.badRequest, err = template.New("bad-request-error").ParseFiles(badRequestFiles...)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
 	return nil
 }
 
+// convertStringToTime formats form time input as time.Time.
+func convertStringToTime(value string) reflect.Value {
+	v, err := time.Parse("2006-01-02", value)
+	if err != nil {
+		log.Println("invalid decoder value")
+		return reflect.Value{}
+	}
+	return reflect.ValueOf(v)
+}
+
+// formToStruct decodes POST form data into a new offer.
+func formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error) {
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+
+	if err := decoder.Decode(&o, req.PostForm); err != nil {
+		return o, err
+	}
+	return o, nil
+}
+
+// CreateOffer handles requests to create new offers.
+func (s *Server) CreateOffer(w http.ResponseWriter, req *http.Request) {
+	o, err := formToStruct(w, req)
+	if err != nil {
+		s.log.Error("failed to convert form to struct", zap.Error(err))
+		s.serveBadRequest(w, req, err)
+		return
+	}
+
+	o.Status = marketing.Active
+	reqType := mux.Vars(req)["offer_type"]
+
+	if reqType == "referral-offer" {
```

Comment:
What if someone puts in `/create/foo`? I would use a switch statement here since we don't want to create `FreeCredit` offer by default.